unit/java_bytecode/java_bytecode_parse_generics/parse_generic_functions.cpp

```diff
@@ -7,69 +7,1015 @@
+          require_type::require_parameter(func_code, "t");
+        require_type::require_pointer(
+          param_t.type(), symbol_typet("java::Generic"));
+
+        THEN("t is generic with type variable T")
+        {
+          require_type::require_java_generic_type(
+            param_t.type(),
+            {{require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+
+          THEN("The bounds are set correctly")
+          {
+            // TODO: the bounds are not parsed yet; extend the tests when
+            // fixed - issue TG-1286
+          }
+        }
+      }
+      THEN("It contains parameter u pointing to Generic")
+      {
+        const code_typet::parametert &param_u =
+          require_type::require_parameter(func_code, "u");
+        require_type::require_pointer(
+          param_u.type(), symbol_typet("java::Generic"));
+
+        THEN("u is generic with type variable U")
+        {
+          require_type::require_java_generic_type(
+            param_u.type(),
+            {{require_type::type_parameter_kindt::Var, class_prefix + "::U"}});
+
+          THEN("The bounds are set correctly")
+          {
+            // TODO: the bounds are not parsed yet; extend the tests when
+            // fixed - issue TG-1286
+          }
+        }
+      }
+    }
+  }
+
+  WHEN("Parsing returnSimple")
+  {
+    THEN("There should be a symbol for the function")
+    {
+      const std::string func_name = ".returnSimple";
+      const std::string func_descriptor = ":()LGeneric;";
+      const std::string process_func_name =
+        class_prefix + func_name + func_descriptor;
+
+      REQUIRE(new_symbol_table.has_symbol(process_func_name));
+
+      const symbolt func_symbol =
+        new_symbol_table.lookup_ref(process_func_name);
+      const code_typet func_code =
+        require_type::require_code(func_symbol.type, 0);
+      THEN("It has return type pointing to Generic")
+      {
+        require_type::require_pointer(
+          func_code.return_type(), symbol_typet("java::Generic"));
+
+        THEN("It is generic with type variable T")
+        {
+          require_type::require_java_generic_type(
+            func_code.return_type(),
+            {{require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+        }
+      }
+    }
+  }
+
+  WHEN("Parsing returnSimpleField")
+  {
+    THEN("There should be a symbol for the function")
+    {
+      const std::string func_name = ".returnSimpleField";
+      const std::string func_descriptor = ":()Ljava/lang/Object;";
+      const std::string process_func_name =
+        class_prefix + func_name + func_descriptor;
+
+      REQUIRE(new_symbol_table.has_symbol(process_func_name));
+
+      const symbolt func_symbol =
+        new_symbol_table.lookup_ref(process_func_name);
+      const code_typet func_code =
+        require_type::require_code(func_symbol.type, 0);
+      // TODO: should point to something else than Object?? - possibly
```

Comment:
Since `T` doesn't have a bound for `returnSimpleField` I think there is no TODO needed here since it is already correctly marked as generic

Reply:
So the `Object` here acts as a bound (i.e., no bound)?